src/sentry/snuba/discover.py

```diff
@@ -2,17 +2,66 @@
 
 import six
 
+from collections import namedtuple
 from copy import deepcopy
-from sentry.api.event_search import get_filter, resolve_field_list
+from datetime import datetime
+
+from sentry.api.event_search import (
+    TAG_KEY_RE,
+    get_filter,
+    resolve_field_list,
+    find_reference_event,
+    InvalidSearchQuery,
+)
+from sentry.snuba.events import get_columns_from_aliases
 from sentry.utils.snuba import (
     Dataset,
+    SnubaTSResult,
     DISCOVER_COLUMN_MAP,
     QUOTED_LITERAL_RE,
     get_function_index,
     raw_query,
     transform_results,
+    zerofill,
 )
 
+ReferenceEvent = namedtuple("ReferenceEvent", ["organization", "slug", "fields"])
+
+
+def create_reference_event_conditions(reference):
```

Comment:
Is this pretty much a straight copy / paste of what's in `get_reference_event_conditions` except with an already looked up event? If so, let's refactor that function to use this one so we don't have 2 versions.